solr/solrj/src/test/org/apache/solr/client/solrj/SolrExampleTests.java

```diff
@@ -724,10 +736,18 @@ public void testStreamingRequest() throws Exception {
     client.commit();
     QueryResponse rsp = client.query( new SolrQuery( "*:*") );
     Assert.assertEquals(0, rsp.getResults().getNumFound());
+    // Create a copy of the file, which can be deleted after uploading.
+    // If the stream isn't closed, the file deletion will fail on Windows,
+    // though it will succeed on linux regardless
```

Comment:
The test is really useful only in Windows? 

Can we just create a FakeInputStream and ensure that `close()` is called?

Reply:
Yes it is. While that can be done for the ContentSteamUpdateRequest, it can't be done for StreamingUpdateRequest without changing the class itself to expose the creation of the steam that it uses... And I didn't want to just go and do that. And it seemed logical to add a test for it, and to use the same logic for both, since they could both suffer from the same problem if changed in the future.

I could refactor both to use something common to stream and close, which would be more easily unit testable?

Reply:
yes, if it is easily unit testable it is better. As most of us develop in non-windows machines, this test becomes essentially useless

Reply:
We do have CI infrastructure that runs on Windows, so a test that only fails on Windows is not useless.  Nonetheless, it's better if it can test everywhere, though not essential in my mind if it's hard in this case.  Your call.